cache/models/subject.js

```diff
@@ -39,6 +39,20 @@ const filters = {
 };
 
 /**
+ * Given absolutePath, return whether the subject is in cache
+ *
+ * @param {String} absolutePath
+ * @returns {Promise} resolves to true for found, false for not
+ */
+function isSubjectInCache(absolutePath) {
+  const subjectKey = sampleStore.toKey('subject', absolutePath);
+
+  // get from cache
+  return redisClient.sismemberAsync(sampleStore.constants.indexKey.subject, subjectKey)
+  .then((exist) => exist);
```

Comment:
Line 52 not needed. Just end line 51 with a semi-colon: you're returning a promise that resolves to 1 (truthy) if the subjectKey is already in the cache.